src/Promitor.Integrations.AzureMonitor/AzureMonitorClient.cs

```diff
@@ -108,9 +110,18 @@ public AzureMonitorClient(AzureEnvironment azureCloud, string tenantId, string s
 
                 // Get the metric value according to the requested aggregation type
                 var requestedMetricAggregate = InterpretMetricValue(aggregationType, mostRecentMetricValue);
-
-                var measuredMetric = string.IsNullOrWhiteSpace(metricDimension) ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate) : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
-                measuredMetrics.Add(measuredMetric);
+                try 
+                {
+                    var measuredMetric = string.IsNullOrWhiteSpace(metricDimension) 
+                                ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate) 
+                                : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
+                    measuredMetrics.Add(measuredMetric);
+                } 
+                catch (MissingDimensionException) 
+                {
+                    _logger.LogWarning("{MetricName} has return a time series with empty value for {Dimension} and the measurements will be dropped", metricName, metricDimension); 
```

Comment:
We might want to use the dimension name from the exception instead of the one available here to make sure we are using the correct one in change we change the code here